web-api/src/storage/elastic.rs

```diff
@@ -497,7 +498,11 @@ struct KnnSearchParts {
 
 impl KnnSearchParams<'_> {
     fn create_common_knn_search_parts(&self) -> KnnSearchParts {
-        let inner_filter = self.create_search_filter();
+        let Ok(Value::Object(inner_filter)) =
+            serde_json::to_value(Clauses::new(self.filter, self.excluded))
+        else {
+            unreachable!(/* filter clauses is valid json object */);
+        };
```

Comment:
📌 in a follow up, we should also apply the same ideas to `KnnSearchParts`, such that we have typed it out completely and the json serialization only happens at the request body creation.